app/services/foreman_ansible/inventory_creator.rb

```diff
@@ -8,11 +8,42 @@ def initialize(hosts)
       @hosts = hosts
     end
 
-    def tempfile
-      tempfile = Tempfile.new("foreman-#{SecureRandom.uuid}-inventory")
-      tempfile.write(hosts.map(&:fqdn).join('\n'))
-      tempfile.close
-      tempfile
+    def json
+      { 'all' => { 'hosts' => hosts.map(&:fqdn) },
+        '_meta' => { 'hostvars' => host_vars } }
+    end
+
+    def host_vars
+      hosts.reduce({}) do |hash, host|
+        hash.update(host.fqdn => { 'foreman' => host_attributes(host),
+                                   'foreman_params' => host_params(host),
+                                   'foreman_ansible_roles' => host_roles(host),
+                                   'foreman_facts' => host_facts(host) })
+      end
+    end
+
+    def host_roles(host)
+      host.all_ansible_roles.map(&:name)
+    end
+
+    def host_attributes(host)
+      render_rabl(host, 'api/v2/hosts/main')
+    end
+
+    def host_params(host)
+      render_rabl(host.host_parameters.authorized, 'api/v2/parameters/base')
+    end
+
+    def host_facts(host)
+      values = FactValue.authorized(:view_facts).my_facts.
+               no_timestamp_facts.where(:host_id => host.id)
+      FactValue.build_facts_hash(values.all)
```

Comment:
The only reason was to mimic the foreman ansible invnetory script, where they do that. I don't insist on that being there. Will remove